lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -2727,6 +2727,11 @@ void ByteCodeGenerator::EmitFunctionBody(FuncInfo *funcInfo)
     }
     Assert(!pnode->CapturesSyms());
     EmitTopLevelStatement(pnode, funcInfo, false);
+
+    if (funcInfo->IsClassConstructor())
+    {
+        funcInfo->GetParsedFunctionBody()->SetHasNoExplicitReturnValue(false);
```

Comment:
I'm guessing that other cases set the flag when they visit the return statement node. I think for class ctors we should either do as Ian suggests and set the flag in StartBindFunction, which is a friendlier place, or have the parser emit an explicit return node.